frame/staking/src/lib.rs

```diff
@@ -1213,23 +1058,72 @@ decl_storage! {
 pub mod migrations {
 	use super::*;
 
-	#[derive(Decode)]
-	struct OldValidatorPrefs {
-		#[codec(compact)]
-		pub commission: Perbill
-	}
-	impl OldValidatorPrefs {
-		fn upgraded(self) -> ValidatorPrefs {
-			ValidatorPrefs {
-				commission: self.commission,
-				.. Default::default()
+	pub mod v6 {
+		use super::*;
+		use frame_support::{traits::Get, weights::Weight, pallet_prelude::*};
+
+		pub trait V6Config: crate::Config {
+			/// The storage prefix of the pallet set by the outer runtime declaration.
+			type PalletPrefix: Get<&'static str>;
```

Comment:
staking is using decl_storage, the prefix is currently fixed as `Staking`, it is not defined at the runtime declaration.

Reply:
Hmm okay then assuming that, I'll change the migration to work based on `on_runtime_upgrade`. This will simplify things a bit,